df_chat/drf/filters.py

```diff
@@ -0,0 +1,21 @@
+from ..models import Message
+from django_filters import rest_framework as filters
+
+
+class MessageFilter(filters.FilterSet):
+    """
+    Filter-set class for `Message` list API.
+    """
+
+    created_gte = filters.DateTimeFilter(field_name="created", lookup_expr="gte")
+    created_lte = filters.DateTimeFilter(field_name="created", lookup_expr="lte")
+
+    username = filters.CharFilter(
+        field_name="room_user__user__username", lookup_expr="contains"
+    )
+
+    body = filters.CharFilter(field_name="body", lookup_expr="contains")
+
+    class Meta:
+        model = Message
+        fields = ()
```

Comment:
Do we have to set the "fields" here to an empty tuple? Could you please check what happens if we remove it completely?

Reply:
Removing it causes an error (Meta classes need to have a `fields` attribute declared).
Also added `icontains`.